modules/nf-core/gsea/gsea/meta.yml

```diff
@@ -0,0 +1,138 @@
+name: "gsea_gsea"
+description: run the Broad Gene Set Enrichment tool in GSEA mode
+keywords:
+  - gene_set_analysis
+tools:
+  - "gsea":
+      description: "Gene Set Enrichment Analysis (GSEA)"
+      homepage: "http://www.gsea-msigdb.org/gsea/index.jsp"
+      documentation: "https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Main_Page"
+      doi: "10.1073/pnas.0506580102"
+      licence: "['BSD-3-clause']"
+
+input:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing information on inputs and contrast, at a minimum
+        an ID, reference and target e.g. [ id:'test', reference:"WT",
+        target:"MUT" ]
+  - gct:
+      type: file
+      description: GCT file with expression values
+      pattern: "*.{gct}"
+  - cls:
+      type: file
+      description: CL file with the classes of the samples in the GCT file
+      pattern: "*.{gct}"
+  - gene_sets:
+      type: file
+      description: GMX or GMT file with gene sets
+      pattern: "*.{gmx,gmt}"
+  - ext.args2:
```

Comment:
Not a channel component! I would rather you say this in `meta` as you've already done, and say how the `meta` info you need should be inserted into the respective args{1,2,3} in `module.config`

Reply:
Ah this is a good point, it is actually on the docs page, just not under guidelines 😅 [^1]

[^1]: https://nf-co.re/docs/contributing/modules#new-module-workflow point 6.2